--- src/dwm/config.def.h	2020-07-25 15:26:45.948553086 -0500
+++ .dotfiles/dwm-config.h	2020-07-25 15:30:53.498546955 -0500
@@ -1,12 +1,13 @@
 /* See LICENSE file for copyright and license details. */
+#include <X11/XF86keysym.h>
 
 /* appearance */
 static const unsigned int borderpx  = 1;        /* border pixel of windows */
 static const unsigned int snap      = 32;       /* snap pixel */
 static const int showbar            = 1;        /* 0 means no bar */
 static const int topbar             = 1;        /* 0 means bottom bar */
-static const char *fonts[]          = { "monospace:size=10" };
-static const char dmenufont[]       = "monospace:size=10";
+static const char *fonts[]          = { "xos4 Terminus:size=10" };
+static const char dmenufont[]       = "xos4 Terminus:size=10";
 static const char col_gray1[]       = "#222222";
 static const char col_gray2[]       = "#444444";
 static const char col_gray3[]       = "#bbbbbb";
@@ -34,12 +35,12 @@ static const Rule rules[] = {
 /* layout(s) */
 static const float mfact     = 0.55; /* factor of master area size [0.05..0.95] */
 static const int nmaster     = 1;    /* number of clients in master area */
-static const int resizehints = 1;    /* 1 means respect size hints in tiled resizals */
+static const int resizehints = 0;    /* 1 means respect size hints in tiled resizals */
 
 static const Layout layouts[] = {
 	/* symbol     arrange function */
-	{ "[]=",      tile },    /* first entry is default */
-	{ "><>",      NULL },    /* no layout function means floating behavior */
+	{ "[T]",      tile },    /* first entry is default */
+	{ "[F]",      NULL },    /* no layout function means floating behavior */
 	{ "[M]",      monocle },
 };
 
@@ -58,12 +59,28 @@ static const Layout layouts[] = {
 static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
 static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
 static const char *termcmd[]  = { "st", NULL };
+static const char *cmdbrightnessup[]  = { "xbacklight", "-inc", "10", NULL };
+static const char *cmdbrightnessdown[]  = { "xbacklight", "-dec", "10", NULL };
+static const char *cmdsoundup[]  = { "pactl", "set-sink-volume", "@DEFAULT_SINK@", "+5%", NULL };
+static const char *cmdsounddown[]  = { "pactl", "set-sink-volume", "@DEFAULT_SINK@", "-5%", NULL };
+static const char *cmdsoundtoggle[] = { "pactl", "set-sink-mute", "@DEFAULT_SINK@", "toggle", NULL };
+static const char *cmdlock[] = { "slock", NULL };
+static const char *cmddisplay[] = { "displayctl", "menu", NULL };
+static const char *cmdnetwork[] = { "networkmanager_dmenu", NULL };
 
 static Key keys[] = {
 	/* modifier                     key        function        argument */
 	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
 	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
 	{ MODKEY,                       XK_b,      togglebar,      {0} },
+  { MODKEY,                       XK_F12,    spawn,          {.v = cmdlock } },
+  { MODKEY,                       XK_o,      spawn,          {.v = cmddisplay } },
+  { MODKEY,                       XK_n,      spawn,          {.v = cmdnetwork } },
+  { 0,                            XF86XK_MonBrightnessDown,     spawn,         {.v = cmdbrightnessdown } },
+  { 0,                            XF86XK_MonBrightnessUp,       spawn,         {.v = cmdbrightnessup } },
+  { 0,                            XF86XK_AudioMute,             spawn,         {.v = cmdsoundtoggle } },
+  { 0,                            XF86XK_AudioRaiseVolume,      spawn,         {.v = cmdsoundup } },
+  { 0,                            XF86XK_AudioLowerVolume,      spawn,         {.v = cmdsounddown } },
 	{ MODKEY,                       XK_j,      focusstack,     {.i = +1 } },
 	{ MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
 	{ MODKEY,                       XK_i,      incnmaster,     {.i = +1 } },
